tests/common/dualtor/dual_tor_io.py

```diff
@@ -475,10 +439,79 @@ def check_tcp_payload(packet, packets_to_send):
                 max(self.lost_packets.items(), key = lambda item:item[1][0:2])
             self.total_disrupt_packets = sum([item[0] for item in self.lost_packets.values()])
             self.total_disrupt_time = sum([item[1] for item in self.lost_packets.values()])
-        elif self.received_counter > 0:
+        elif self.total_lost_packets == 0:
             self.max_lost_id = 0
             self.max_disrupt_time = 0
             self.total_disrupt_packets = 0
             self.total_disrupt_time = 0
             logger.info("Gaps in forwarding not found.")
+
         logger.info("Packet flow examine finished after {}".format(str(datetime.datetime.now() - examine_start)))
+        logger.info("Total number of filtered incoming packets captured {}".format(self.received_counter))
+        logger.info("Number of duplicated packets received: {}".format(self.duplicated_packets_count))
+        logger.info("Number of packets lost: {}".format(self.total_lost_packets))
+
+
+    def examine_each_packet(self, packets):
+        lost_packets = dict()
+        sent_packets = dict()
+        duplicated_packets_count = 0
+        prev_payload, prev_time = None, None
+        sent_payload = 0
+        disruption_start, disruption_stop = None, None
+        received_counter = 0    # Counts packets from dut.
+        for packet in packets:
+            if packet[scapyall.Ether].dst == self.sent_pkt_dst_mac:
+                # This is a sent packet - keep track of it as payload_id:timestamp.
+                sent_payload = int(str(packet[scapyall.TCP].payload).replace('X',''))
+                sent_packets[sent_payload] = packet.time
+                continue
+            if packet[scapyall.Ether].src in self.received_pkt_src_mac:
+                # This is a received packet.
+                received_time = packet.time
+                received_payload = int(str(packet[scapyall.TCP].payload).replace('X',''))
+                if received_payload == prev_payload:
+                    # make account for packet duplication, and keep looking for a new and unique received packet
```

Comment:
I don't think we can guarantee that duplicated packets (especially coming from different ToRs) will arrive sequentially - it could be that ToR A can send the duplicate packet plus some other packet, then ToR B sends the duplicate packet again. Maybe change to `<=`?

Reply:
We sort these packets (before examining) with sequential payload as the sorting key.
https://github.com/Azure/sonic-mgmt/blob/59e87d63f2fc3a2188467adefa2a7f0d0d3d0467/tests/common/dualtor/dual_tor_io.py#L421

Reply:
Ah I missed that, thanks!